engine.go

```diff
@@ -220,6 +223,29 @@ func (e *RTCEngine) handleLocalTrackPublished(res *livekit.TrackPublishedRespons
 	e.trackPublishedChan <- res
 }
 
+func (e *RTCEngine) handleLossyDataPacket(msg webrtc.DataChannelMessage) {
+	packet, err := e.ReadDataPacket(msg)
+	if err != nil {
+		return
+	}
+	switch msg := packet.Value.(type) {
+	case *livekit.DataPacket_Speaker:
+		if e.OnActiveSpeakersChanged != nil {
+			e.OnActiveSpeakersChanged(msg.Speaker.Speakers)
+		}
+	}
+}
+
+func (e *RTCEngine) ReadDataPacket(msg webrtc.DataChannelMessage) (*livekit.DataPacket, error) {
```

Comment:
this is probably better to be package private.

```suggestion
func (e *RTCEngine) readDataPacket(msg webrtc.DataChannelMessage) (*livekit.DataPacket, error) {
```

Reply:
I've made it private thus